src/step_definitions/api.js

```diff
@@ -19,4 +28,12 @@ defineSupportCode(function({ When, Then }) {
   Then(/^the response should exact match to body:$/, function(body) {
     return expect(fetchResult.hasBodyMatch(JSON.parse(body))).to.be.true;
   });
+
+  Then(/^the response should match JSON schema:$/, function(schema) {
+    try {
+      fetchResult.hasMatchingSchema(JSON.parse(schema));
+    } catch (error) {
+      console.error(error);
```

Comment:
This step should fail if the JSON schema is not matched, now you got just error displayed in console. 
In other words, you should `return Promise.reject(error)`